### PR TITLE
fix(TDI-39500) : add support of SOQL Relationship queries

### PR DESCRIPTION
**Pull request by akhabali**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-39500

**What is the new behavior?**
add support of SOQL Relationship queries

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
Need to be merged only after #985

### CODE REVIEW

**Comment by akhabali** [comment]
Need to be merged only after #986

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1294/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1294/)

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/BulkResultSet.java [hunk 1]
```diff
@@ -33,7 +33,10 @@ public BulkResult next() throws IOException {
             if ((row = reader.getValues()) != null) {
                 result = new BulkResult();
                 for (int i = 0; i < this.header.size(); i++) {
-                    result.setValue(header.get(i), row[i]);
+                    //We replace the . with _ to add support of relationShip Queries
+                    //The relationShip Queries Use . in Salesforce and we use _ in Talend (Studio)
+                    //So Account.Name in SF will be Account_Name in Talend
+                    result.setValue(header.get(i).replace('.', '_'), row[i]);
```
Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/BulkResultSet.java]: Is there any tests for this case?

Comment (akhabali) [reply components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/BulkResultSet.java]: test add in 175ffd2fbd6405f956d508c4559eebfcb9f9a622

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1303/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1303/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aakhabali%2FTDI-39500-SOQL_Relationship_support)

**Pull request merged**